File: components/credentials/src/main/java/org/cloudfoundry/credhub/requests/RsaSetRequest.java
package org.cloudfoundry.credhub.requests;

import java.util.Objects;

import javax.validation.Valid;
import javax.validation.constraints.NotNull;

import com.fasterxml.jackson.annotation.JsonProperty;
import org.cloudfoundry.credhub.ErrorMessages;
import org.cloudfoundry.credhub.credential.RsaCredentialValue;

public class RsaSetRequest extends BaseCredentialSetRequest<RsaCredentialValue> {

  @NotNull(message = ErrorMessages.MISSING_VALUE)
  @Valid
  @JsonProperty("value")
  private RsaCredentialValue rsaKeyValue;

  public RsaCredentialValue getRsaKeyValue() {
    return rsaKeyValue;
  }

  public void setRsaKeyValue(final RsaCredentialValue rsaKeyValue) {
    this.rsaKeyValue = rsaKeyValue;
  }

  @Override
  public RsaCredentialValue getCredentialValue() {
    return rsaKeyValue;
  }

  @Override
  public GenerationParameters getGenerationParameters() {
    return null;
  }

  @Override
  public boolean equals(final Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    final RsaSetRequest that = (RsaSetRequest) o;
    return Objects.equals(rsaKeyValue, that.rsaKeyValue);
  }

  @Override
  public int hashCode() {
    return Objects.hash(rsaKeyValue);
  }
}

File: components/credentials/src/main/kotlin/org/cloudfoundry/credhub/requests/RsaSetRequest.kt
package org.cloudfoundry.credhub.requests

import com.fasterxml.jackson.annotation.JsonProperty
import org.cloudfoundry.credhub.ErrorMessages
import org.cloudfoundry.credhub.credential.RsaCredentialValue
import java.util.Objects
import javax.validation.Valid
import javax.validation.constraints.NotNull

class RsaSetRequest : BaseCredentialSetRequest<RsaCredentialValue?>() {
    @NotNull(message = ErrorMessages.MISSING_VALUE)
    @Valid
    @JsonProperty("value")
    var rsaKeyValue: RsaCredentialValue? = null

    override val generationParameters: GenerationParameters?
        get() = null

    override val credentialValue: RsaCredentialValue?
        get() {
            return rsaKeyValue
        }

    override fun equals(other: Any?): Boolean {
        if (this === other) {
            return true
        }
        if (other == null || javaClass != other.javaClass) {
            return false
        }
        val that = other as RsaSetRequest?
        return rsaKeyValue == that!!.rsaKeyValue
    }

    override fun hashCode(): Int = Objects.hash(rsaKeyValue)
}
